examples/help_usage.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2017-2023, University of Cincinnati, developed by Henry Schreiner
+// under NSF AWARD 1414736 and by the respective contributors.
+// All rights reserved.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <CLI/CLI.hpp>
+#include <string>
+
+int main(int argc, char **argv) {
+    std::string input_file_name, output_file_name;
+    int level, subopt;
+
+    // app caption
+    CLI::App app{"CLI11 help"};
+
+    app.require_subcommand(1);
+    // subcommands options and flags
+    CLI::App *const encode = app.add_subcommand("e", "encode")->ignore_case();  // ignore case
+    encode->add_option("input", input_file_name, "input file")
+        ->option_text(" ")
+        ->required()
+        ->check(CLI::ExistingFile);                                                               // file must exist
+    encode->add_option("output", output_file_name, "output file")->option_text(" ")->required();  // required option
+    encode->add_option("-l, --level", level, "encoding level")
+        ->option_text("[1..9]")
+        ->check(CLI::Range(1, 9))
+        ->default_val(5);                                            // limit parameter range
+    encode->add_option("-R, --remove", level, "remove input file");  // no parameter option
```

Comment:
level here is used in another option, should probably be add_flag so it doesn't default to requiring an additional argument.   this matches your other comment